spyder/api/shellconnect/main_widget.py

```diff
@@ -111,16 +109,42 @@ def remove_shellwidget(self, shellwidget):
             self.update_actions()
 
     def set_shellwidget(self, shellwidget):
-        """
-        Set widget associated with shellwidget as the current widget.
-        """
+        """Set widget associated with shellwidget as the current widget."""
         old_widget = self.current_widget()
         widget = self.get_widget_for_shellwidget(shellwidget)
         if widget is None:
             return
         self._stack.setCurrentWidget(widget)
-        self.switch_widget(widget, old_widget)
-        self.update_actions()
+
+        # This prevents errors when the widget we're switching to is a
+        # PaneEmptyWidget
+        try:
+            self.switch_widget(widget, old_widget)
+            self.update_actions()
+        except AttributeError:
+            pass
+
+    def add_errored_shellwidget(self, shellwidget):
+        """
+        Create a new PaneEmptyWidget in the stack and associate it to
+        shellwidget.
+
+        This is necessary to show a meaningful message when switching to
+        consoles with dead kernels.
+        """
+        shellwidget_id = id(shellwidget)
+        if shellwidget_id not in self._shellwidgets:
+            widget = PaneEmptyWidget(
+                self,
+                "variable-explorer",  # TODO: Use custom icon here
```

Comment:
Is the icon to be used here still being in the process of being defined, right? Probably the `TODO` will need to be done in a follow-up PR? or maybe you want to address that here?

Reply:
> Is the icon to be used here still being in the process of being defined, right?

Yep, although I think we already have one thanks to @conradolandia.

> Probably the TODO will need to be done in a follow-up PR? or maybe you want to address that here?

It'll be addressed in a follow-up PR by @conradolandia.